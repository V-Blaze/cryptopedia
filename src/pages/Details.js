import React from 'react';
import { useLocation } from 'react-router-dom';

const Details = () => {
  const location = useLocation();
  const {
    exchangeId, name, rank, percentTotalVolume, volumeUsd, tradingPairs, socket, exchangeUrl,
  } = location.state;
  return (
    <div>
      <div>
        <img src={`https://logo.clearbit.com/${exchangeUrl}`} alt="logo" />
        <div>
          <span>
            name:
            {name}
          </span>
          <span>
            rank
            {rank}
          </span>
        </div>
      </div>
      <div>
        <h3>Exchange Details</h3>
        <ul>
          <li>
            <span>Trading Volum percent</span>
            <span>
              {percentTotalVolume}
              %
            </span>
          </li>
          <li>
            <span>Trading Volum USD</span>
            <span>
              {volumeUsd}
              $
            </span>
          </li>
          <li>
            <span>No. of Trading Pairs</span>
            <span>{tradingPairs}</span>
          </li>
          <li>
            <span>Socket</span>
            <span>{socket}</span>
          </li>
          <li>
            <span>Website</span>
            <span>{exchangeUrl}</span>
          </li>
          <li>
            <span>Exchange ID</span>
            <span>{exchangeId}</span>
          </li>
        </ul>
      </div>
    </div>
  );
};

export default Details;
