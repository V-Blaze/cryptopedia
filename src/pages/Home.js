import React, { useState, useEffect } from 'react';
import { useSelector } from 'react-redux';
import { useNavigate } from 'react-router-dom';
import ExchangeLists from '../components/ExchangeLists';
import { filterCountry } from '../utils';

const Home = () => {
  const navigate = useNavigate();
  const exchanges = useSelector((state) => state.exchanges);
  const [cryptoExchanges, setCryptoExchanges] = useState([]);

  useEffect(() => {
    setCryptoExchanges(exchanges);
  }, [exchanges]);

  const handleSelection = (country) => {
    const filtered = filterCountry(country, exchanges);
    setCryptoExchanges(filtered);
  };

  const handleSearch = (text) => {
    const filteredExchanges = exchanges.filter((ex) => (
      ex.name.toLowerCase().includes(text.toLowerCase())));
    setCryptoExchanges(filteredExchanges);
  };

  const showDetails = (id) => {
    const item = exchanges.filter((exchange) => exchange.exchangeId === id);
    navigate('/details', { state: item[0] });
    console.clear();
  };

  return (
    <div>
      <div className="bg-[#3775e8] h-[15rem] flex flex-col gap-3 justify-center items-center">
        <div>
          <span className=" text-xl font-semibold italic text-gray-300 ">
            Total Exchanges:
            {`  ${cryptoExchanges.length}`}
          </span>
        </div>
        <div className="w-full flex flex-col items-center gap-2">
          <input
            type="text"
            placeholder="Search Exchange"
            onChange={(e) => handleSearch(e.target.value)}
            className="w-[80%] rounded-lg px-2 py-2 bg-blue-900 text-white"
          />
          <form className="w-[60%]">
            {/* <label htmlFor="countries">Choose an countries:</label> */}
            <select
              id="countries"
              name="countries"
              onChange={(e) => handleSelection(e.target.value)}
              className="w-full rounded-md p-1 bg-blue-900 text-white"
            >
              <option value="Select A Country">Select A Country</option>
              <option value="allCountries">All Countries</option>
              <option value="Australia">Australia</option>
              <option value="China">China</option>
              <option value="EU">EU</option>
              <option value="Indonesia">Indonesia</option>
              <option value="Japan">Japan</option>
              <option value="Russia">Russia</option>
              <option value="Seychelles">Seychelles</option>
              <option value="Singapore">Singapore</option>
              <option value="South_Korea">South Korea</option>
              <option value="UK">UK</option>
              <option value="US">US</option>
              <option value="otherCountries">Other Countries</option>
            </select>
          </form>
        </div>
      </div>
      <div className="exchange-container mb-9 ">
        {cryptoExchanges.map((exchange) => (
          <ExchangeLists
            key={exchange.exchangeId}
            exchange={exchange}
            showDetails={showDetails}
          />
        ))}
      </div>
      <div className=" fixed bottom-0 mt-4 w-full bg-[#4369B2] h-9 flex justify-center items-center text-white text-xl">
        <span>CryptoPedia Copywrite V-Blaze</span>
      </div>
    </div>
  );
};

export default Home;
