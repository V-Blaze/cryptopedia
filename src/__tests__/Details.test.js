import { screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from '../redux/configureStore';
import { Details } from '../pages';
import renderWithProviders from '../redux/Exchanges/exchange-test-utils';

const mockHistoryPush = jest.fn();
jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useHistory: () => ({
    push: mockHistoryPush,
  }),
  useLocation: () => ({
    pathname: '/details',
    state: { name: 'Binance' },
  }),
}));

it('snapshots', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <BrowserRouter>
          <Details />
        </BrowserRouter>
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('testing the heading inside home component', async () => {
  renderWithProviders(
    <BrowserRouter>
      <Details />
    </BrowserRouter>,
  );

  // should show no user initially, and not be fetching a user
  expect(screen.getByText(/Exchange Details/i)).toBeInTheDocument();
  expect(screen.getByText(/Binance/i)).toBeInTheDocument();
});
