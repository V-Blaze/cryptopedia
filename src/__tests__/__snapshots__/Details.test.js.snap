// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshots 1`] = `
<div>
  <div>
    <img
      alt="logo"
      src="https://logo.clearbit.com/undefined"
    />
    <div>
      <span>
        name:
        Binance
      </span>
      <span>
        rank
      </span>
    </div>
  </div>
  <div>
    <h3>
      Exchange Details
    </h3>
    <ul>
      <li>
        <span>
          Trading Volum percent
        </span>
        <span>
          %
        </span>
      </li>
      <li>
        <span>
          Trading Volum USD
        </span>
        <span>
          $
        </span>
      </li>
      <li>
        <span>
          No. of Trading Pairs
        </span>
        <span />
      </li>
      <li>
        <span>
          Socket
        </span>
        <span />
      </li>
      <li>
        <span>
          Website
        </span>
        <span />
      </li>
      <li>
        <span>
          Exchange ID
        </span>
        <span />
      </li>
    </ul>
  </div>
</div>
`;
